### ISSUE TITLE
Support non-literal-URL QR codes in the integrated QR code scanner, including parsing "MEBKM" format.

### ISSUE THREAD

**Issue by kufue** [fork-maintainers/iceraven-browser#118]
I tried to scan a qr bookmark in another app, and it said that no app can open it, iceraven-browser has its own QR code scanner but I couldn't check if it works there because it can read only from camera and not from storage. That looks like another feature request.

### What is the user problem or growth opportunity you want to see solved?
Everywhere we go, there should be a safe way to communicate, because we must protect ourselves from desieses
### How do you know that this problem exists today? Why is this important?
Many companies don't allow their workers to be close to customers because of CoV
### Who will benefit from it?
Everyone, as it is a protection layer, that doesn't allow unnecessary crowds, while sharing the most important link for a place. It doesn't mean one must wait for it to load to then bookmark the site, it means fastly write down contact information

**Comment by ghost** [comment]
It's most likely, if anything, to come from upstream Fenix. A better option would be to just use a dedicated QR scanner app. The only reason Fenix and Iceraven have one is to sync between devices.

**Comment by interfect** [comment]
So you have an *image* of a QR code stored on the device, and you want to... share the image at the browser, have the browser pull the QR code out of the picture, and then go to the link?

**Comment by interfect** [comment]
No, I figured it out. You have a QR code that encodes, literally, something like `MEBKM:TITLE:Main Page - Wikipedia, the free encyclopedia;URL:http://en.wikipedia.org/wiki/Main_Page;;`.


![image](https://user-images.githubusercontent.com/752054/93417287-92b16380-f85c-11ea-8859-2ebfaee9d421.png)

These are out in the wild for Covid purposes. ZXing Barcode Scanner can scan this, work out that it refers to a URL and also ships the title (like a bookmark in a browser), and give you a button to go to the URL. The browser-integrated scanner can't handle it and just plugs the whole string into the default search engine.

It would be nice if we could parse this format and go to the URL. We'd also probably want to try and contribute the feature upstream; I bet they'd take it.

**Comment by interfect** [comment]
I'd recommend opening an [issue on Mozilla Firefox](https://github.com/mozilla-mobile/fenix/issues/new/choose) about this. Maybe they'll fix it and we won't have to.
